internal/server/manage_restrictions.go

```diff
@@ -252,6 +252,13 @@ func ManageRestrictions(client ManageRestrictionsClient, manageTmpl template.Tem
 						return err
 					}
 
+					if taskID != 0 {
+						err := client.ClearTask(ctx, taskID)
+						if handleError(w, &data, err) {
+							return err
+						}
+					}
```

Comment:
I'd suggest putting this in the API to keep the business logic together. Also because it will do it in a DB transaction so if anything fails it will rollback, whereas putting the logic here means you could end up editing the severance data but not clearing the task if the second request failed.